spock-specs/src/test/groovy/org/spockframework/smoke/Interceptors.groovy

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.smoke
+
+import org.spockframework.EmbeddedSpecification
+import org.spockframework.runtime.extension.ExtensionAnnotation
+import org.spockframework.runtime.extension.IAnnotationDrivenExtension
+import org.spockframework.runtime.extension.IMethodInvocation
+import org.spockframework.runtime.model.SpecInfo
+import org.spockframework.specs.extension.Snapshot
+import org.spockframework.specs.extension.Snapshotter
+import spock.lang.ResourceLock
+
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+
+class Interceptors extends EmbeddedSpecification {
+  @Snapshot
+  Snapshotter snapshotter
+
+  def setup() {
+    runner.addClassImport(LifecycleTest)
+    LifecycleRecorderAndContextTesterExtension.lifecycleOutline = ''
+  }
+
+  @ResourceLock("LifecycleRecorderAndContextTesterExtension.lifecycleOutline")
+  def "interceptors are called in the correct order"() {
+    when:
+    runner.runWithImports """
+abstract class SuperSpec extends Specification {
+  @Shared superShared = 0
+
+  def superInstance = 0
+
+  def setupSpec() {
+  }
+
+  def setup() {
+  }
+
+  def superFeature1() {
+    expect: true
+  }
+
+  def superFeature2() {
+    expect: true
+  }
+
+  def superDataFeature1() {
+    expect: x
+    where:
+    x << [true, true]
+  }
+
+  def superDataFeature2() {
+    expect: x
+    where:
+    x << [true, true]
+  }
+
+  def cleanup() {
+  }
+
+  def cleanupSpec() {
+  }
+}
+
+@LifecycleTest
+class SubSpec extends SuperSpec {
+  @Shared subShared = 0
+
+  def subInstance = 0
+
+  def setupSpec() {
+  }
+
+  def setup() {
+  }
+
+  def subFeature1() {
+    expect: true
+  }
+
+  def subFeature2() {
+    expect: true
+  }
+
+  def subDataFeature1() {
+    expect: x
+    where:
+    x << [true, true]
+  }
+
+  def subDataFeature2() {
+    expect: x
+    where:
+    x << [true, true]
+  }
+
+  def cleanup() {
+  }
+
+  def cleanupSpec() {
+  }
+}
+"""
+
+    then:
+    noExceptionThrown()
```

Comment:
Isn't this implied, normally it is only used if there is nothing else to assert on

Reply:
Probably, no idea why I added it in the first place